src/main/java/in/projecteka/consentmanager/common/cache/ICacheAdapter.java

```diff
@@ -0,0 +1,11 @@
+package in.projecteka.consentmanager.common.cache;
+
+public interface ICacheAdapter<K,V> {
+    V get(K key);
+
+    void put(K key, V value);
+
+    V getIfPresent(K value);
+
+    void invalidate(K key);
+}
```

Comment:
Instead of defining our own abstraction, can we use spring cache abstraction if possible?

Reply:
Implementing spring's cache interfaces might be a bit of overkill for now. We mostly would have to implement methods that we don't use currently or leave them without implementations.

Reply:
I don't think you need to implement everything if you use Cache Manager. But That's okay you can go ahead with your implementation